public/app/features/dashboard-scene/settings/AnnotationsEditView.tsx

```diff
@@ -36,58 +37,136 @@ export class AnnotationsEditView extends SceneObjectBase<AnnotationsEditViewStat
     return getDashboardSceneFor(this);
   }
 
-  private get _dataLayers(): SceneDataLayerProvider[] {
-    return sceneGraph.getDataLayers(this._dashboard);
-  }
-
-  public getSceneDataLayers(): SceneDataLayers | undefined {
+  public getSceneDataLayers(): SceneDataLayers {
     const data = sceneGraph.getData(this);
 
     if (!(data instanceof SceneDataLayers)) {
-      return undefined;
+      throw new Error('SceneDataLayers not found');
     }
 
     return data;
   }
 
   public getAnnotationsLength(): number {
-    return this._dataLayers.filter((layer) => layer.topic === DataTopic.Annotations).length;
+    return this.getSceneDataLayers().state.layers.filter((layer) => layer.topic === DataTopic.Annotations).length;
   }
 
   public getDashboard(): DashboardScene {
     return this._dashboard;
   }
 
   public onNew = () => {
-    console.log('todo: onNew');
+    const newAnnotationQuery: AnnotationQuery = {
+      name: newAnnotationName,
+      enable: true,
+      datasource: getDataSourceRef(getDataSourceSrv().getInstanceSettings(null)!),
+      iconColor: 'red',
+    };
+
+    const newAnnotation = new DashboardAnnotationsDataLayer({
+      key: `annotations-${newAnnotationQuery.name}`,
+      query: newAnnotationQuery,
+      name: newAnnotationQuery.name,
+      isEnabled: Boolean(newAnnotationQuery.enable),
+      isHidden: Boolean(newAnnotationQuery.hide),
+    });
+
+    const data = this.getSceneDataLayers();
+
+    if (data) {
```

Comment:
This actually needs to be changed since we're now throwing an error if there are no scene data layers